include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -987,54 +987,57 @@ struct __scan
 // __brick_includes
 //------------------------------------------------------------------------
 
-template <typename _Compare, typename _Size1, typename _Size2>
+template <typename _Size1, typename _Size2, typename _Compare, typename _Proj1 = oneapi::dpl::identity,
+          typename _Proj2 = oneapi::dpl::identity>
 struct __brick_includes
 {
+    _Size1 __size1;
+    _Size2 __size2;
     _Compare __comp;
-    _Size1 __na;
-    _Size2 __nb;
+    _Proj1 __proj1;
+    _Proj2 __proj2;
 
-    __brick_includes(_Compare __c, _Size1 __n1, _Size2 __n2) : __comp(__c), __na(__n1), __nb(__n2) {}
+    __brick_includes(_Size1 __size1, _Size2 __size2, _Compare __comp, _Proj1 __proj1 = {}, _Proj2 __proj2 = {})
+        : __size1(__size1), __size2(__size2), __comp(__comp), __proj1(__proj1), __proj2(__proj2)
+    {
+    }
 
-    template <typename _ItemId, typename _Acc1, typename _Acc2>
+    template <typename _ItemId, typename __Rng1, typename __Rng2>
     bool
-    operator()(_ItemId __idx, const _Acc1& __b_acc, const _Acc2& __a_acc) const
+    operator()(_ItemId __idx, const __Rng1& __rng1, const __Rng2& __rng2) const
     {
-        using ::std::get;
-
-        auto __a = __a_acc;
-        auto __b = __b_acc;
-
-        auto __a_beg = _Size1(0);
-        auto __a_end = __na;
+        using std::get;
 
-        auto __b_beg = _Size2(0);
-        auto __b_end = __nb;
+        auto __proj_rng_1 = [this](auto&& __val) { return std::invoke(__proj1, std::forward<decltype(__val)>(__val)); };
```

Comment:
Is this lambda usage problematic for the multiple kernels issue? Should they be made into member functions instead?